cpp/include/cudf/strings/strings_column_handler.hpp

```diff
@@ -0,0 +1,127 @@
+  size_type count() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal char data array
+   *---------------------------------------------------------------------------**/
+  const char* chars_data() const;
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal offsets array
+   *---------------------------------------------------------------------------**/
+  const int32_t* offsets_data() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the size of the char data array in bytes
+   *---------------------------------------------------------------------------**/
+  size_type chars_column_size() const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a pointer to the internal null mask memory
+   *---------------------------------------------------------------------------**/
+  const bitmask_type* null_mask() const;
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the number of nulls in this column
+   *---------------------------------------------------------------------------**/
+  size_type null_count() const;
+
+  enum sort_type {
+        none=0,    ///< no sorting
+        length=1,  ///< sort by string length
+        name=2     ///< sort by characters code-points
+    };
+
+  /**---------------------------------------------------------------------------*
+   * @brief Prints the strings to stdout.
+   * 
+   * @param start Index of first string to print.
+   * @param end Index of last string to print. Specify -1 for all strings.
+   * @param max_width Maximum number of characters to print per string.
+   *                  Specify -1 to print all characters.
+   * @param delimiter The chars to print between each string.
+   *                  Default is new-line character.
+   *---------------------------------------------------------------------------**/
+  void print( size_type start=0, size_type end=-1,
+              size_type max_width=-1, const char* delimiter = "\n" ) const;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a new strings column created from a subset of
+   * of this instance's strings column.
+   * 
+   * @param start Index of first string to use.
+   * @param end Index of last string to use.
+   * @param step Increment value between indexes.
+   * @param stream CUDA stream to use kernels in this method.
+   * @return New strings column of size (end-start)/step.
+   *---------------------------------------------------------------------------**/
+  std::unique_ptr<cudf::column> sublist( size_type start, size_type end, size_type step, cudaStream_t stream=0 );
```

Comment:
Make these free functions that act on `strings_column_handler` objects. The conversion from `column_view` to `strings_column_handler` should be implicit.

Reply:
This makes me a bit nervous placing function names like `gather` and `sort` into the `cudf` namespace. They have the first parameter `strings_column_handler` but any `column_view` can be automatically converted.

Reply:
Yeah, I'd suggest putting them in the `cudf::strings` namespace.